src/framework/net/httplogin.cpp

```diff
@@ -136,6 +142,63 @@ void LoginHttp::httpLogin(const std::string& host, const std::string& path,
             });
         }
     });
+#else
+    g_asyncDispatcher.detach_task(
+        [this, host, path, port, email, password, request_id, httpLogin] {
+        emscripten_fetch_attr_t attr;
+        emscripten_fetch_attr_init(&attr);
+        strcpy(attr.requestMethod, "POST");
+        static const char* const headers[] = {
+            "Content-Type", "application/json; charset=utf-8",
+            0,
+        };
+        attr.requestHeaders = headers;
+        attr.attributes = EMSCRIPTEN_FETCH_LOAD_TO_MEMORY | EMSCRIPTEN_FETCH_SYNCHRONOUS;
+        std::string body = "{\"email\":\"" + email + "\",\"password\":\"" + password + "\",\"stayloggedin\":true,\"type\":\"login\"}";
```

Comment:
i mean, this only works as long as the password does not include any of `\` or `/` or `"`  or `'`

so technically this is wrong, but to fix it "properly" we need to introduce a proper JSON api/library, but as a quickfix we can do something like
```c++
std::string jsonEncodeString(const std::string_view input) {
    std::ostringstream encoded;
    
    for (char c : input) {
        switch (c) {
            case '"':  encoded << "\\\""; break;   // Escape double quotes
            case '\\': encoded << "\\\\"; break;   // Escape backslash
            case '\b': encoded << "\\b"; break;    // Escape backspace
            case '\f': encoded << "\\f"; break;    // Escape form feed
            case '\n': encoded << "\\n"; break;    // Escape new line
            case '\r': encoded << "\\r"; break;    // Escape carriage return
            case '\t': encoded << "\\t"; break;    // Escape tab
            default:
                // Check if the character is a control character or non-printable ASCII
                if (static_cast<unsigned char>(c) < 0x20 || static_cast<unsigned char>(c) > 0x7F) {
                    // Encode using Unicode escape
                    encoded << "\\u"
                            << std::hex << std::uppercase
                            << static_cast<int>(static_cast<unsigned char>(c) >> 4)
                            << static_cast<int>(static_cast<unsigned char>(c) & 0xF);
                } else {
                    encoded << c;  // Leave printable characters unchanged
                }
                break;
        }
    }
    
    return encoded.str();
}
```
and
```c++
[jsonEncodeString](std::string body = "{\"email\":" + jsonEncodeString(email) + ",\"password\":" + jsonEncodeString(password) + ",\"stayloggedin\":true,\"type\":\"login\"}";)
```

Reply:
On line 78 there is the original implementation using:
`json body = json{ {"email", email}, {"password", password}, {"stayloggedin", true}, {"type", "login"} };`

Should I use it instead?

Reply:
@OTArchive great, yes go for it 👍